frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -0,0 +1,171 @@
+
+const { ThinSpace } = Unicode;
+
+type Props = {
+  onCancel: (...args: Array<any>) => any;
+  isOpen: boolean;
+  tracingId: any;
+  tracingStoreUrl: any;
+  visibleSegmentationLayer: any;
+  relevantSegments: Segment[];
+  parentGroup: number;
+  groupTree: TreeNode[];
+};
+
+type SegmentInfo = {
+  segmentId: number;
+  segmentName: string;
+  groupId: number | undefined | null;
+  groupName: string;
+  volumeInNm3: number;
+  formattedSize: string;
+  volumeInVoxel: number;
+  boundingBoxTopLeft: Vector3;
+  boundingBoxPosition: Vector3;
+};
+
+const exportStatisticsToCSV = (
+  segmentInformation: Array<SegmentInfo>,
+  tracingId: string,
+  groupId: number,
+) => {
+  if (segmentInformation.length < 0) {
+    return;
+  }
+  const segmentStatisticsAsString = segmentInformation
+    .map(
+      (segmentInfo) =>
+        `${segmentInfo.segmentId},${segmentInfo.segmentName},${segmentInfo.groupId},${segmentInfo.groupName},${segmentInfo.volumeInVoxel},${segmentInfo.volumeInNm3},${segmentInfo.boundingBoxTopLeft},${segmentInfo.boundingBoxPosition}`,
+    )
+    .join("\n");
+  const csv = [SEGMENT_STATISTICS_CSV_HEADER, segmentStatisticsAsString].join("\n");
+  const filename = `segmentStatistics_tracing-${tracingId}_group-${groupId}.csv`;
+  const blob = new Blob([csv], {
+    type: "text/plain;charset=utf-8",
+  });
+  saveAs(blob, filename);
+};
+
+export function SegmentStatisticsModal({
+  isOpen,
+  onCancel,
+  tracingId,
+  tracingStoreUrl,
+  visibleSegmentationLayer,
+  relevantSegments: segments,
+  parentGroup,
+  groupTree,
+}: Props) {
+  const mag = getResolutionInfo(visibleSegmentationLayer.resolutions);
+  const nmFactorToUnit = new Map([[1, "nm³"]]);
+  const dataSource = useFetch(
+    async () => {
+      await new Promise((resolve) => setTimeout(resolve, 10000)); //TODO delete; only for testing
+      await api.tracing.save();
+      let resultObjects = await Promise.all([
+        getSegmentVolumes(
+          tracingStoreUrl,
+          tracingId,
+          mag.getHighestResolution(),
+          segments.map((segment) => segment.id),
+        ),
+        getSegmentBoundingBoxes(
+          tracingStoreUrl,
+          tracingId,
+          mag.getHighestResolution(),
+          segments.map((segment) => segment.id),
+        ),
+      ]).then((response) => {
+        const segmentSizes = response[0];
+        const boundingBoxes = response[1];
+        const statisticsObjects = [];
+        for (let i = 0; i < segments.length; i++) {
+          // segments in request and their statistics in response are in the same order
+          const currentSegment = segments[i];
+          const currentBoundingBox = boundingBoxes[i];
+          const segmentStatObject = {
+            segmentId: currentSegment.id,
+            segmentName:
+              currentSegment.name == null ? `Segment ${currentSegment.id}` : currentSegment.name,
+            groupId: currentSegment.groupId == null ? -1 : currentSegment.groupId,
```

Comment:
I know that there could be cases where the groupId is null though the segment is not in the root group. due to the lack of time today, I didn't implement a better way to find this out, but I can still add a function to find this out!